### DISCUSSION TITLE
What's the best way to fix these false detections?

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#13558] [General Support] [answered]
### Describe the problem you are having

Hello

In this specific case with a trained F+ model in place and illustrated with screenshots  further down the page, is it better to 

1 - play with aspect ratio or area for an actual dog when seen in that FOV such that the foliage error falls outside of those ranges?   In other words, use `min_area` and `max_area`  by adding a min_area / max_area filter, or by playing with object proportions based on `min_ratio` and `max_ratio`?

2 - finely adjust the thresholds in the config (the figures are tongue in cheek to show that it could get very fiddly!)
```
objects:
  filters:
    dog:
      min_score: .84552
      threshold: .92034
```
or

3 -  use an object filter mask regarding that troublesome area of foliage ?

or

4 - something else?

Having read the documentation I think the first option is the best way to go but that is quite a process to implement and I want to check with experts before I start. My trained model is working well and all object detections have been "elevated" so that I can set thresholds at (say) 85-90% in order to filter out false positives.

But I do have a few peculiar detections still ... for example

I have got dozens and dozens of these, which you can see is foliage rather than a dog:
![image](https://github.com/user-attachments/assets/b7f6f491-c910-410b-9ce0-2226d4e43fa2)

But I also have even more of these which show an actual dog:
![image](https://github.com/user-attachments/assets/6a197116-651d-490d-8ac0-71fabedd58c4)

The foliage shows up as typically 92-25% dog whereas an actual dog in the same camera view is a bit higher.  There is some overlap between the top of the lower range and the bottom of the higher range but anyway, the ranges are very close to one another.

I can try some experiments with aspect ratio and detected image size but there might be overlap (e.g. a small image of a dog lower down the steps might be the same size as the foliage misdetection and hence I think the actual dog would be ignored).

What's your experience been - how finely tuned do the filters need to be in order to discriminate whilst being reliable?

### Version

0.13 and 0.14

### What browser(s) are you using?

n/a

### Frigate config file

```yaml
objects:
  filters:
    dog:
      min_score: .8
      threshold: .9
    cat:
      min_score: .65
…
```

### Relevant Frigate log output

```shell
n/a
```

### Relevant go2rtc log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Frigate stats

```json
n/a
```

### Operating system

Debian

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Mixed

### Camera make and model

n/a

### Screenshots of the Frigate UI's System metrics pages

n/a

### Any other information that may be helpful

n/a

**Comment by NickM-27** [comment, marked as answer]
you will definitely want to submit the first example as false positives, it may take a good number but it should improve results. Otherwise, seems like an object mask would be a good solution

**Reply by ElectronicBattle** [reply]
Excellent - I have already submitted about eighty of the false "foliage" ones (and perhaps ten true examples of actual  dogs on the same camera).  In that case I'll ask for a new model in due course, perhaps when I have submitted a hundred false positive foliage examples.  

I presume that also submitting some true positives does help train the next model; is that the case? 

If that doesn't do the job, I'll then experiment with an object mask.

**Reply by NickM-27** [reply]
do keep in mind that true positives are important, only submitting false positives won't lead to as good of results.

**Reply by ElectronicBattle** [reply]
I thought as much. I have always done  something in the region of 80/20  true/false % leading up to my first model and now I'll do the same continuing towards the next one.  I recall https://docs.frigate.video/plus/first_model which says 

> It is recommended to submit both true positives and false positives. This will help the model differentiate between what is and isn't correct. You should aim for a target of 80% true positive submissions and 20% false positives across all of your images. If you are experiencing false positives in a specific area, submitting true positives for any object type near that area in similar lighting conditions will help teach the model what that area looks like when no objects are present.

This must clearly apply to first *and* subsequent models.  Which sponsors another question: will my next model presumably build upon my first model? I'd already submitted hundreds of dog "true positives" in that camera view so the 80/20 true/false is probably "ready" if so.
